ERCS/erc-7604.md

```diff
@@ -0,0 +1,202 @@
+---
+eip: 7604
+title: ERC-1155 Permit Approvals
+description: Permit approvals for ERC-1155 tokens
+author: calvbore (@calvbore), emiliolanzalaco (@emiliolanzalaco)
+discussions-to: https://ethereum-magicians.org/t/proposal-for-a-new-eip-erc-2612-style-permits-for-erc1155-nfts/15504
+status: Draft
+type: Standards Track
+category: ERC
+created: 2024-01-27
+requires: 165, 712, 1155, 5216
+---
+
+## Abstract
+
+The "permit" approval flow for both [ERC-20](./erc-20.md) and [ERC-721](./erc-721.md) are large improvements for the existing UX of the token underlying each ERC. This ERC extends the "permit" pattern to [ERC-1155](./erc-20.md) tokens, borrowing heavily upon both [ERC-4494](./erc-4494.md) and [ERC-2612](./erc-2612.md).
+
+The structure of [ERC-1155](./erc-1155.md) tokens requires a new ERC to account for the token standard's use of both token IDs and balances (also why this ERC requires [ERC-5216](./erc-5216.md)). An additional field of arbitrary `bytes` is added for extra data to be added to the signatures in the case that the permit function would be extended in some way.  
```

Comment:
I am personally not a fan of the inclusion of an arbitrary `bytes` data field. 
I feel it's unnecessary and out of scope of the requirements of a permit function - 99.99% of the time it will be a waste of gas (about 350 units).
It's also not present in any other permit standards

Reply:
I am open to removing this, but I would also like to hear some more feedback on possible usecases, or if it is mostly useless, before making a final decision.

Reply:
@emiliolanzalaco I think I have removed all the references to the additional `bytes data` paramater. Can you give the doc a look over to make sure I didn't miss any?